core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/ResourceURIWithQuery.java

```diff
@@ -64,8 +64,16 @@ public ResourceURIWithQuery(String uri) {
 	
 	ResourceURIWithQuery(ResourceURI resourceUri, String query) {
 		this.resourceUri = resourceUri;
-		this.query = query;
-		this.uri = Strings.isNullOrEmpty(query) ? resourceUri.toString() : String.join(QUERY_PART_SEPARATOR, resourceUri.toString(), query);
+		if (Strings.isNullOrEmpty(query)) {
+			this.query = query;
+			this.uri = resourceUri.toString();
+		} else if (query.startsWith(QUERY_PART_SEPARATOR)) {
+			this.query = query.substring(1);
+			this.uri = String.join("", resourceUri.toString(), query);
```

Comment:
What was the reason for introducing this check?

Reply:
Double question mark in URIs.